cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/DistributedProgramRunnableModule.java

```diff
@@ -83,7 +86,7 @@ public DistributedProgramRunnableModule(CConfiguration cConf, Configuration hCon
   }
 
   // usable from any program runtime, such as mapreduce task, spark task, etc
-  public Module createModule() {
+  public Module createModule(@Nullable final ProgramId programId, @Nullable final String principal) {
```

Comment:
This is also called from DistributedMapReduceTaskContextProvider, SparkRuntimeContextProvider which does not need to pass this information correct ?